sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -540,10 +564,25 @@ object DottyPlugin extends AutoPlugin {
       dottyVersion,
       fullLoader,
       libraryLoader,
-      Array(dottyLibrary, scalaLibrary),
+      libraryJars,
       compiler,
       all.toArray,
       None)
+  }
+}
+
+private class TopClassLoader(sbtLoader: ClassLoader) extends ClassLoader(null) {
+  private val sharedPrefixes = List(
+    "xsbti.",
+    "org.jline."
+  )
 
+  override protected def loadClass(name: String, resolve: Boolean): Class[_] = {
+    if (sharedPrefixes.exists(name.startsWith(_))) {
```

Comment:
Since this is run on every class we load, I think it's worth optimizing this as much as we can:
```suggestion
    if (name.startsWith("xsbti.") || name.startsWith("org.jline."))) {
```